### DISCUSSION TITLE
Avoiding runtime compilation

### DISCUSSION THREAD

**Discussion by miezys** [evrete/evrete#58] [Q&A] [answered]
Hello,

we are using evrete with annotated java rule classes (`JAVA-CLASS`). Is there any way to avoid compilation at runtime, so we could run our application in a jvm without the need for a jdk?

Thanks in advance

**Comment by andbi** [comment, marked as answer]
Yes, it should be possible. Runtime compilation only comes into play when there's a literal condition or action. Both can be rewritten to use functional interfaces instead. 

When using `JAVA_CLASS` sources, you can use `@MethodPredicate` inside the corresponding `@Where` annotations as described in https://evrete.org/docs/ajr/

**Reply by miezys** [reply]
Thanks for prompt reply, it did helped!
